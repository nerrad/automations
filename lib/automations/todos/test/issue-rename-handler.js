const { gimmeOctokit, gimmeContext, gimmeApp } = require( '@testHelpers' );

describe( 'issue-rename-handler', () => {
	let octokit, context, app;
	beforeEach( () => {
		octokit = gimmeOctokit();
		context = gimmeContext( 'issues', 'edited' );
		app = gimmeApp( 'todos' );
	} );
	it( 'un-edits the issue title', async () => {
		await app( context, octokit );
		expect( octokit.issues.update.mock.calls[ 0 ][ 0 ] ).toMatchSnapshot();
		expect(
			octokit.issues.createComment.mock.calls[ 0 ][ 0 ]
		).toMatchSnapshot();
	} );
	it( 'only acts if the title is edited', async () => {
		context.payload.changes = {};
		await app( context, octokit );
		expect( octokit.issues.update ).not.toHaveBeenCalled();
		expect( octokit.issues.createComment ).not.toHaveBeenCalled();
	} );
} );
