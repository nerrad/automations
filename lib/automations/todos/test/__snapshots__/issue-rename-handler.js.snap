// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue-rename-handler un-edits the issue title 1`] = `
Object {
  "issue_number": 35,
  "owner": "nerrad",
  "repo": "tests",
  "title": "I have been edited",
}
`;

exports[`issue-rename-handler un-edits the issue title 2`] = `
Object {
  "body": ":rocket: Please do not change this issue title! **The todo action** uses it to prevent duplicate issues from being opened. If you think this is an error, please [open an issue](https://github.com/nerrad/automations/issues/new)!",
  "issue_number": 35,
  "owner": "nerrad",
  "repo": "tests",
}
`;
